apps/emqx_bridge_gcp_pubsub/src/emqx_bridge_gcp_pubsub_connector.erl

```diff
@@ -76,11 +88,22 @@ on_start(
     }),
     %% emulating the emulator behavior
     %% https://cloud.google.com/pubsub/docs/emulator
-    HostPort = os:getenv("PUBSUB_EMULATOR_HOST", "pubsub.googleapis.com:443"),
+    {Transport, HostPort} =
+        case os:getenv("PUBSUB_EMULATOR_HOST") of
+            false ->
+                {tls, "pubsub.googleapis.com:443"};
+            HostPort0 ->
+                %% The emulator is plain HTTP...
+                Transport0 = persistent_term:get({?MODULE, transport}, tcp),
+                {Transport0, HostPort0}
+        end,
```

Comment:
Nit: the code seems to be quite overloaded with different kinds injection (through env, through persistent term).

Maybe introduce some kind of injectable `emqx_constant`

```
{Transport, HostPort} = emqx_constant:get(gcp_pubsub, {tls, "pubsub.googleapis.com:443"})
```

or just move to a separate fun...

Reply:
:heavy_check_mark: